### PR TITLE
Add remaining instances for Comonad transformers

### LINKED ISSUE #35: Add remaining Comonad* instances

### PR DESCRIPTION
**Pull request by skeate**
**Description of the change**

Adds `ComonadAsk`, `ComonadEnv`, and `ComonadTraced` instances for `StoreT`, `EnvT`, and `TracedT` 

Closes #35

---

**Checklist:**

- [x] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0000)")
- [x] Linked any existing issues or proposals that this pull request should close
- [x] Updated or added relevant documentation
- [x] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by JordanMartinez** [comment]
:ping_pong: @garyb

**Review by JordanMartinez — APPROVED** [review]

**Comment by garyb** [comment]
I did take a look at this, but I don't feel qualified to approve it, since I'm not that familiar with comonad patterns. Perhaps @MonoidMusician or @LiamGoodacre could chime in, I think they're more theorypilled than me? :wink:

**Review by MonoidMusician — APPROVED** [review]
Yup, looks good to me! It’s doing the “obvious” thing, which is just the opposite of the usual monadic constructions.

**Comment by garyb** [comment]
> Yup, looks good to me! It’s doing the “obvious” thing, which is just the opposite of the usual monadic constructions.

Ah okay. I did once implement some comonad stuff and it seemed way too trivial to be correct. I asked someone and they weren't sure if it was right either. :smile:

**Comment by MonoidMusician** [comment]
Yeah, this is just lifting instances like `ask = lift <<< ask` in the monadic work, but it's `ask = ask <<< lower` here :)

**Comment by MonoidMusician** [comment]
Are we good to merge this btw?

File: src/Control/Comonad/Traced/Class.purs [hunk 1]
```diff
@@ -47,3 +51,15 @@ censor f (TracedT tr) = TracedT ((f >>> _) <$> tr)
 
 instance comonadTracedTracedT :: (Comonad w, Monoid t) => ComonadTraced t (TracedT t w) where
   track t (TracedT tr) = extract tr t
+
+lowerTrack :: forall t m w a. ComonadTrans t => ComonadTraced m w => m -> t w a -> a
+lowerTrack m = track m <<< lower
```
Comment (LiamGoodacre) [inline src/Control/Comonad/Traced/Class.purs:56]: is this an implementation detail, or something to be exported? (It's currently being implicitly exported).

Comment (skeate) [reply src/Control/Comonad/Traced/Class.purs:56]: whoops, yeah this should not be exported

**Pull request merged**